File: longest_strictly_increasing_decreasing_subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        current = 1
        result = 1
        increase = 0
        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                if increase > 0:
                    current += 1
                else:
                    current = 2
                    increase = 1
            elif nums[i - 1] > nums[i]:
                if increase < 0:
                    current += 1
                else:
                    current = 2
                    increase = -1
            else:
                current = 1
                increase = 0
                
            result = max(result, current)
            
        return result

File: longestStrictlyIncreasingDecreasingSubarray.ts
function longestMonotonicSubarray(nums: number[]): number {
    let current: number = 1
    let result: number = 1
    let increase: number = 0
    
    for(let i = 1; i < nums.length; i++){
        if(nums[i - 1] < nums[i]){
            if(increase > 0){
                current++
            }else{
                current = 2
                increase = 1
            }
        }else if(nums[i - 1] > nums[i]){
            if(increase < 0){
                current++
            }else{
                current = 2
                increase = -1
            }
        }else{
            current = 1
            increase = 0
        }
        result = Math.max(current, result)
    }
    return result
}